### ISSUE TITLE
class scope fixture is recognized as function scope

### ISSUE THREAD

**Issue by reverseark** [pytest-dev/pytest#3941]
[Edit: Bug repro is further simplified. Please see the repro code below]

Hi, I found a case when class scope fixture is recognized as function scope, which fails the test with error "You tried to access the 'function' scoped fixture '...' with a 'class' scoped request object"

Could you please confirm if this is a pytest bug, or if this is not an encouraged pattern and I should try to implement it differently?

platform win32 -- Python 3.6.6, pytest-3.7.3, py-1.5.4, pluggy-0.7.1

```
@pytest.fixture(scope="class")
def class_dep(request):
    print("Fixture setup")

@pytest.fixture(scope="function")
def func_dep():
    pass

@pytest.mark.usefixtures("class_dep")
@pytest.mark.parametrize('class_dep', [10,20], indirect=True)
class TestScenario():
    def test_foo(self, func_dep):
        pass
    
    def test_bar(self):
        pass


> pytest.exe "test_foo1.py::TestScenario" -v -s

test_foo1.py::TestScenario::test_foo[10] Fixture setup
PASSED
test_foo1.py::TestScenario::test_foo[20] Fixture setup
PASSED
test_foo1.py::TestScenario::test_bar[10] Fixture setup
PASSED
test_foo1.py::TestScenario::test_bar[20] Fixture setup
PASSED
```

**"Fixture setup" is printed 4 times.**
If I change def test_foo(self, func_dep): to drop reference to func_dep, then it works as expected ("Fixture setup is printed 2 times").

**Comment by RonnyPfannschmidt** [comment]
The bug is that parametrize overrides the scope, this should break at collection

**Comment by reverseark** [comment]
Hi @RonnyPfannschmidt, the repro code is further simplifed. It looks like the parametrize for class fixture is totally broken.

**Comment by nicoddemus** [comment]
Hi @reverseark,

Add `scope="class"` to your parametrize definition, then it works:

```
.tmp\test_class_fix_scope.py::TestScenario::test_foo[10] Fixture setup
PASSED
.tmp\test_class_fix_scope.py::TestScenario::test_bar[10] PASSED
.tmp\test_class_fix_scope.py::TestScenario::test_foo[20] Fixture setup
PASSED
.tmp\test_class_fix_scope.py::TestScenario::test_bar[20] PASSED
``` 

But it seems to be a bug `parametrize` is not using `class_dep`'s scope automatically.

**Comment by nicoddemus** [comment]
Hmm yep, there's a bug in the function which finds the parametrization scope when it is not explicitly given. It takes all fixtures used by the function, not only those in the parametrization definition, which causes it to look at `class_dep` and `func_dep` used by `test_foo` and decide that it should use the narrower scope, `function`.

**Comment by reverseark** [comment]
Hi @nicoddemus, thanks a lot for the explanation. My knowledge around parametrize is limited and am glad to learn explicit scope setting is actually supported for parametrize, as a very neat workaround in this case.

**Comment by nicoddemus** [comment]
Sure thing @reverseark, thanks for taking the time to narrow down the example, we always appreciate that. PR with fix in #3987.

**Comment by MikeRomanenko** [comment]
Had to debug this issue for a few hours until found this thread, thanks a bunch. I know it is already resolved.
Pytest version: 6.2.3
Issue: Parametrize mark decorator defaults to 'function' scope.
